dask/base.py

```diff
@@ -172,9 +172,9 @@ def normalize_array(x):
                 data = md5(b'-'.join([str(item).encode() for item in x.flat])).hexdigest()
         else:
             try:
-                data = md5(x.view('i1').data).hexdigest()
+                data = md5(x.ravel().view('i1').data).hexdigest()
```

Comment:
Might I suggest using `.ravel(order='K')` instead? That will avoid a copy for other non-contiguous arrays when possible (though the overhead is probably nominal compared to the hash function)